### PR TITLE
[CM-586] Do not resolve uid2 when running in privacy mode

### PR DESCRIPTION
**Pull request by mschuwalow**
https://liveintent.atlassian.net/browse/CM-586

Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

### CODE REVIEW

**Review by 3link — APPROVED** [review]

**Comment by 3link** [comment]
Like the refactoring. 👍

**Review by leonelcuevas — APPROVED** [review]

File: src/idex/common.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    return {
+      resolve: (successCallback, errorCallback, additionalParams) => {
+        try {
+          unsafeResolve(successCallback, errorCallback, additionalParams)
+        } catch (e) {
+          console.error('IdentityResolve', e)
+          errorCallback()
+          fromError('IdentityResolve', e)
```
[the remark this answers is not in the document]
Comment (mschuwalow) [reply src/idex/common.js:160]: I guess so? :D Kept them the same as they were.

The string passed to `fromError` is what shows up here https://prod-grafana.liveintent.com/d/3WtwayWWz/livepixel?orgId=2&refresh=10s&viewPanel=13, I don't know whether the value has any meaning beyond it

**Comment by mschuwalow** [comment]
Talked with Dejan and we are going to make a change in idres that will make the nonId/unifiedId handling cleaner. Will keep this unmerged until then

File: src/idex/common.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+/**
+ * @type {Cache}
+ */
+export const noopCache = {
+  get: (key) => {
+    return null
+  },
+  set: (key, value) => {
+
+  }
```
Comment (wi101) [inline src/idex/common.js:61]: I would do it this way, but as you prefer
```suggestion
  get: (key) => null,
  set: (key, value) => undefined
```

File: src/idex/common.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      tuples.push(asStringParam(retrievedIdentifier.name, retrievedIdentifier.value))
+    })
+
+    const attributeResolutionAllowed = (attribute) => {
+      if (attribute === 'uid2') {
+        return !config.privacyMode
+      } else {
+        return true
+      }
+    }
```
Comment (wi101) [inline src/idex/common.js:102]:
```suggestion
    const attributeResolutionAllowed = (attribute) => {
      attribute === 'uid2' ? !config.privacyMode : true
    }
```

**Review by wi101 — APPROVED** [review]
LGTM, just tiny suggestions

**Review by 3link — APPROVED** [review]

**Pull request merged**